server/server.js

```diff
@@ -41,19 +41,28 @@ async function handleValidRequest(button, numPresses, batteryLevel) {
     let client = await db.beginTransaction()
 
     let session = await db.getUnrespondedSessionWithButtonId(button.button_id, client)
-        
-    if (session === null) {
-        session = await db.createSession(button.installation_id, button.button_id, button.unit, button.phone_number, numPresses, client)
-    }
-    else {
-        session.incrementButtonPresses(numPresses)
-        await db.saveSession(session, client)
-    }
 
     if ((batteryLevel !== undefined) &&
             (batteryLevel >= 0) &&
             (batteryLevel <= 100)) {
-        await db.saveButtonBatteryLevel(button.button_serial_number, batteryLevel, client)
+        if (session === null) {
```

Comment:
This looks like it'll work, but I feel like the logic could be cleaned up somewhat to make it clearer and reduce repetition. Maybe by doing the logic to know whether we have a good batteryLevel or not first and then you can use that to do the create or update based on whether sessions is null.

Reply:
do you mean assigning the battery level by itself and then doing the null check?

Reply:
Yes, something like that. But no big deal if you don't want to fiddle with this anymore.

Reply:
I went to do this but then realized that I would need to change more than just this bit to ensure that we don't update the sessions table with a null value and overwrite the last useful battery value (if for some reason one press sends the battery level and the next one doesn't). So I'm leaving this as is for now.

Reply:
OK. That's fine. Thanks for looking into it.